tests/functional/kmip/highlevel.spec.js

```diff
@@ -120,4 +122,130 @@ describe('KMIP High Level Driver', () => {
         });
     });
+
+    it('should retry without kmipHandshake on thales session revoked', async () => {
+        const options = {
+            kmip: {
+                providerName: 'tests',
+                client: {
+                    bucketNameAttributeName: null,
+                    compoundCreateActivate: false,
+                },
+                codec: {},
+                transport: {
+                    pipelineDepth: 8,
+                    tls: {
+                        port: 5696,
+                    },
+                },
+            },
+        };
+        const kmipClient = new KMIPClient(options, TTLVCodec, LoopbackServerTransport);
+        // healthcheck to start the connection and the kmipHandshake
+        void await util.promisify(kmipClient.healthcheck.bind(kmipClient))(logger);
+
+        const kmsKey = await util.promisify(kmipClient.createBucketKey.bind(kmipClient))('mybucket', logger);
+
+        const encrypted = await util.promisify(kmipClient.cipherDataKey.bind(kmipClient))(
+            1, kmsKey, Buffer.from('abc'), logger);
+
+        const reason = 'General Failure';
+        const message = '[NCERRUnauthorizedAccess]: Token has been revoked';
+
+        // the next call should return a thales session revoked error only once
+        const stub = sinon.stub(kmipClient.kmip.transport.channel, '_transform')
+            .onFirstCall()
+            .callsFake((_cipherFunc, _request, cb) => {
+                kmipClient.kmip.transport.channel.stubbedResult = true;
+                cb(null, kmipClient.kmip.transport.channel.errorResponse(reason, message));
+                kmipClient.kmip.transport.channel.stubbedResult = false;
+            })
+            .callThrough();
+        const tlsWriteSpy = sinon.spy(kmipClient.kmip.transport.channel, 'write');
+        const tlsEndSpy = sinon.spy(kmipClient.kmip.transport.channel, 'end');
+        const tlsConnectSpy = sinon.spy(kmipClient.kmip.transport.channel, 'connect');
+
+
+        const encrypted2 = await util.promisify(kmipClient.cipherDataKey.bind(kmipClient))(
+            1, kmsKey, Buffer.from('abc'), logger);
+
+        assert.strictEqual(Buffer.compare(encrypted, encrypted2), 0,
+            'second call to cipherDataKey with a retry should match 1st call',
+        );
+        assert.strictEqual(tlsEndSpy.callCount, 1,
+            `tls connection should have been ended once instead of ${tlsEndSpy.callCount}`);
+
+        assert.strictEqual(tlsConnectSpy.callCount, 1,
+            `tls connect should have been called once instead of ${tlsConnectSpy.callCount}`);
+
+        assert.strictEqual(tlsWriteSpy.callCount, 2,
+            `tls write should have been called twice because of retry instead of ${tlsWriteSpy.callCount}`);
+        assert.strictEqual(Buffer.compare(tlsWriteSpy.getCall(0).args[0], tlsWriteSpy.getCall(1).args[0]), 0,
+            'tls write should have twice the same argument');
+        assert.strictEqual(stub.callCount, 2,
```

Comment:
Do we plan to increase this count in the future?

Reply:
No, at this place we don't expect any configurable retry. It's only 1 specific message we know we can retry immediately otherwise any other error message should be sent back to client